SIPS/sip-29.md

```diff
@@ -136,35 +134,59 @@ type AssetConversionRate = {
 
   // The UNIX timestamp of when the conversion rate will expire.
   expirationTime?: number;
-
-  // Market data for the asset pair.
-  // The values of the type string are expected to represent decimal numbers in a string.
-  marketData?: {
-    marketCap: string;
-    totalVolume: string;
-    circulatingSupply: string;
-    allTimeHigh: string;
-    allTimeLow: string;
-    pricePercentChange: {
-      // The `all` value is a special interval that represents all available data.  
-      all?: number;  
-
-      // The interval key MUST follow the ISO 8601 duration format.  
-      [interval: string]: number;
-    };
-  };
 };
 
+
 type FromAsset = Conversion["from"];
 
 type ToAsset = Conversion["to"];
 
 type OnAssetsConversionResponse = {
-  conversionRates: Record<From, Record<To, AssetConversionRate | null>>;
+  conversionRates: Record<From, Record<To, AssetConversion | null>>;
+};
+```
+
+#### Get Assets market data
+
+```typescript
+import { OnAssetsMarketDataHandler } from "@metamask/snaps-sdk";
+
+export const onAssetsMarketData: OnAssetsMarketDataHandler = async ({
+  assets
+}) => {
+  const marketData = /* Get market data for given `assets` */;
+  return { marketData };
+};
+```
+
+The type for an `onAssetsMarketData` handler function’s arguments is:
+
+```typescript
+type AssetPair = {
+  from: Caip19AssetTypeOrId;
```

Comment:
IMO, we shouldn’t reuse `AssetPair` in the arguments of `onAssetMarketData`. The terms from and to don’t make much sense in this context since there is no conversion involved.

Something like `{ asset, unit }` would probably be more explicit.